src/main/java/plansoft/skillsmanagement/level/LevelService.java

```diff
@@ -24,8 +24,21 @@
 	}
 
 	public void addLevel(Level level) {
+		List<Level> levels = new ArrayList<>();
+		levelRepository.findAll().forEach(levels::add);
+		for (int i = 0; i < levels.size(); i++) {
+			if (levels.get(i).getLevelOrder() == level.getLevelOrder()) {
+				int currentLevelOrder = level.getLevelOrder();
+				for (int j = 0; j < levels.size(); j++) {
```

Comment:
Anche qui come sopra non utilizzi l'indice dell'elemento in maniera attiva, ma solo per eseguire il retrive dell'elemento dalla lista quindi potresti usare il foreach